### ISSUE TITLE
old data over MQTT

### ISSUE THREAD

**Issue by ChristophCaina** [snaptec/openWB#2283]
I am using the openWB and pulling different information via MQTT to use them in my HomeAssistant Setup.
This will be done by the openWB integration itself (https://github.com/a529987659852/openwbmqtt) and also with NodeRed.

Now, I have noticed some strange behave with the values that are delivered over MQTT.

on a regular basis, when I reboot one of the systems, the values are getting reset to some older values - which have been updated already.

For example:
The last time I have charged my Car, the total amount of charged kWh was near 160 kWh in total.
since then, the value has been reset to 81 kWh ...
![grafik](https://user-images.githubusercontent.com/26391061/177582512-136b6df7-89d7-4baa-8a18-d4d481bf2e9e.png)
![grafik](https://user-images.githubusercontent.com/26391061/177582733-7fe65280-ea40-49db-a95d-90a122b84362.png)

The interesting thing is, that it will always reset to this value... I have seen it several times between different charges.
![grafik](https://user-images.githubusercontent.com/26391061/177583312-d85abda7-2367-4c82-87c8-f59b3b95aa99.png)

I think, this might be somehow because the OpenWB is not regularely sending out the same Topics when there's no change - and somehow older values have been stored "somewhere"... similar like a cache?
I've noticed this behave with the Integration I've linked above... and tried to workaround this with a node red configuration which connects to the OpenWB directly.

Unfortunately, after it seems to work as expected, today the behave is completely the same - and the values are the same like in the OpenWB HomeAssistant integration.
![grafik](https://user-images.githubusercontent.com/26391061/177583745-77210476-f6f1-4874-80ba-a3d350953221.png)

![grafik](https://user-images.githubusercontent.com/26391061/177583766-a6fad348-3e77-436f-ab46-4c504d824c96.png)

![grafik](https://user-images.githubusercontent.com/26391061/177583790-43f447d9-cac4-4b13-971d-c911fdfff12e.png)

Open WB Software Version is 1.9.271

**Comment by dj3mu** [comment]
The "caching" is most likely in the MQTT-Server itself which persists data (infinite) according to the setting of the "Retain" flag of the MQTT message.

To get latest values, try sending a message with raw content `1` to MQTT topic `openWB/set/RenewMQTT`. It should force the openWB to retransmit all MQTT values.

**Comment by dj3mu** [comment]
First of all, please also check if the data is really re-transmitted.

Second, please don't send that message every 10 seconds. It will cause significant additional load to the openWB.  
Ideally, you should only and exactly send it once your nodered is restarted.  
Or in which other cases do you encounter this problem?

**Comment by ChristophCaina** [comment]
I am not quite sure... probably only when HomeAssistant / NodeRed is restarting.
I will do further tests, but sending the message seemed to work

**Comment by ChristophCaina** [comment]
Hi, I need to bring that topic up again.
In the setup mentioned above - I had a MQTT Broker acting as a bridge.
this MQTT broker has received the data from the MQTT Broker of the OpenWB.

I did some further research and noticed, that - in my case - the old values came from my own Broker (at least in the cases that I have observed recently)...

Since then, I've changed the setup.
I am now connected directly to the MQTT Broker in the Wallbox and have removed the bridge in between.

First tests showed, that everything seemed to be OK now - restarts didn't show any old meter values from the OpenWB.

But today - I noticed, that the same situation happened again yesterday evening.
My Smarthome system received an old Meter-Value...
After round about 5 min, the value switched back to the correct value.

Here's how it looked in the history of my Smarthome Software:

![grafik](https://user-images.githubusercontent.com/26391061/196653796-c14cd4ba-85b7-494c-9259-3854fe9f2b2e.png)
![grafik](https://user-images.githubusercontent.com/26391061/196653816-33da2082-e1d0-4d61-bc7b-074fb2d679ed.png)
![grafik](https://user-images.githubusercontent.com/26391061/196653845-1329f6d7-15c6-427c-90ed-70b2184b430f.png)
![grafik](https://user-images.githubusercontent.com/26391061/196653863-4d24e63e-dd4c-4a1e-ba4a-d0d5e0af5126.png)

due to the unexpected occurence, I don't have any logs from the OpenWB available - but I will try to do some further research.
At that time, I am not aware of any reboot of the OpenWB during that time frame...

Other people reported similar issues in the forum:
![Uploading grafik.png…]()
